### DISCUSSION TITLE
Ctrl + click on function definition with type overload does not show references

### DISCUSSION THREAD

**Discussion by diego-pm** [microsoft/pylance-release#3472] [Enhancement]
When pressing ctrl + click on a function definition, vs code will show all places where the function is called, which is great.
However, if the function has overloaded type definitions, this won't work and instead, vs code will only show the overload definitions.

Example code:
```python
from __future__ import annotations
from typing import overload


@overload
def foo(a: int, b: int = ...) -> str:
    ...
    
@overload
def foo(a: str, b: int = ...) -> int:
    ...

def foo(a: int | str, b: int = 3) -> int | str:
    if isinstance(a, int):
        return str(a + b)
    else:
        return int(a) + b
    
    
if __name__ == '__main__':
    foo(1)
    foo('2')
```

When pressing ctrl+click on the function definition, references are not shown:
![image](https://user-images.githubusercontent.com/99246556/195564543-7a5068a4-7e50-4812-9b92-a146747943e7.png)

But it would be great if it behaved as if there were no overloads, where references are shown:
![image](https://user-images.githubusercontent.com/99246556/195564739-5682387e-d91c-4fdf-bf04-f96d30eefcbb.png)

Could vs code also show references when ctrl + click on a function definition, even if it has type overload?

**Comment by rchiodo** [comment]
I believe this is a function of how VS code behaves. I can see it in the messages they send to pylance.

When you ctrl+click, VS code asks us for the definitions for the thing you're hovering over. 

Then depending upon that result, it either asks for all references (if there's one) or shows the list of the definitions.

So I believe this would be a request on VS code
